en/dev_setup/dev_env_windows_wsl.md

```diff
@@ -0,0 +1,161 @@
+- Direct serial MCU flashing doesn't work<br>
+  Flashing can be done through QGC (see [instructions below](dev_env_windows_wsl.md#flash-a-flight-control-board))
+
+### Requirements for Graphics Support
+
+The requirement for any out of the box graphics support in WSL is [WSLg](https://github.com/microsoft/wslg).
+It is included in WSL on all recent enough Windows versions namely all Windows 11 versions and a few recent enough insider Windows 10 builds.
+Check the linked project GitHub page to read the latest availability.
+If these requirements are not met then you can still build PX4, flash it to boards using QGC for Windows and run the simulation in headless mode. All windowed, graphics features like e.g. gazebo rendered visualization, QGC running in WSL, git gui, will not work without [WSLg](https://github.com/microsoft/wslg).
+
+## Installation Instructions
+
+### Install WSL2
+
+To install WSL2 with the default Ubuntu distribution on a new installation of Windows 11:
+
+1. Open _cmd.exe_ as administrator.
+   Can be done by pressing the start key, typing "cmd", right klicking on the _Command prompt_ entry and selecting **Run as administrator**.
+1. Execute the command `wsl --install` to run the installation routine for WSL.
+1. Reboot the system.
+1. Open `cmd` again as a normal user (not as administrator).
+   This can be done by pressing the **Start** key, typing "cmd" and pressing **Enter**.
+1. Execute the command `wsl` to access the WSL shell.
+1. WSL will prompt you for a user name and password for the Ubuntu installation.
+   **Note:** _Remember the credentials for later on!_
+
+:::note
+If you have any problems with your setup, check the current microsoft WSL installation documentation.
+:::
+
+### Install PX4 Toolchain
+
+To install the development toolchain for Gazebo simulation, JMAVSim simulation and Pixhawk/NuttX harware:
+
+1. Execute the command `cd ~` to switch to the home folder of WSL for the next steps.
+   :::warning
+   This is important!
+   If you work from a location outside of the WSL file system you'll run into issues such as very slow execution and access right/permission errors.
+   :::
+
+1. [Download PX4 Source Code](../dev_setup/building_px4.md):
+   ```bash
+   git clone https://github.com/PX4/PX4-Autopilot.git --recursive
+   ```
+1. Run the **ubuntu.sh** installer script and acknowledge any prompts as the script progresses:
+   ```bash
+   bash ./PX4-Autopilot/Tools/setup/ubuntu.sh
+   ```
+   
+   :::note
+   This installs tools to build PX4 for Pixhawk, Gazebo and JMAVSim targets.
+   - You can use the `--no-nuttx` and `--no-sim-tools` options to omit the NuttX and/or simulation tools.
+   - Other Linux build targets are untested (you can try these by entering the appropriate commands in [Ubuntu Development Environment](../dev_setup/dev_env_linux_ubuntu.md) into the WSL shell.
+     Note that `git` is already installed in WSL.
+   
+1. Restart the "WSL computer" on completion.
+   ```
+   wsl --shutdown
```

Comment:
FYI `wsl` and `wsl --shutdown` don't work for me:
```
Command 'wsl' not found, but can be installed with:
sudo apt install wsl
```
Doesn't seem to matter - I just exit and restart.

Reply:
Sorry if it wasn't written clearly. The commands `wsl --shutdown` and `wsl` are to be ran in a native Windows cmd shell. So if the WSL shell was accessed through cmd after typing `exit` on the WSL bash and returning to the original shell.

Reply:
So this fails
```
wsl
wsl --shutdown
```
but this works
```
wsl
exit
wsl --shutdown
```
So I think you are saying you can only run wsl-shutdown after exiting? in the raw cmd prompt?

Reply:
Crystal clear. I moved the open command earlier and I only refer to this way of doing things now.